src/services/invitation/service.ts

```diff
@@ -136,4 +140,98 @@ export class InvitationService {
     const memberships = invitations.map(({ permission, item }) => ({ item, member, permission }));
     await itemMembershipRepository.createMany(memberships);
   }
+
+  async handleCSVInvitations(
+    actor: Actor,
+    repositories: Repositories,
+    query: IdParam,
+    file: MultipartFile | undefined,
+  ) {
+    const { id: parentId, template_id } = query;
+    const parentItem = await repositories.itemRepository.get(parentId);
+    await validatePermission(repositories, PermissionLevel.Admin, actor, parentItem);
+
+    if (!file || file.mimetype != CSV_MIMETYPE) {
+      throw new Error(`
+          An incorrect type of file has been uploaded, 
+          Please upload a file with .csv extension
+        `);
+    }
```

Comment:
Moving this check as the first thing in the function above will improve performance, as you should not even try to validate the item existence if you did not get a file.